pkg/planner/core/task.go

```diff
@@ -861,7 +875,23 @@ func (p *NominalSort) Attach2Task(tasks ...base.Task) base.Task {
 	return t
 }
 
-func (p *PhysicalTopN) getPushedDownTopN(childPlan base.PhysicalPlan) *PhysicalTopN {
+func (p *PhysicalTopN) getPushedDownTopN(childPlan base.PhysicalPlan, storeTp kv.StoreType) (*PhysicalTopN, *PhysicalTopN) {
+	var newGlobalTopN *PhysicalTopN
+
+	fixValue := fixcontrol.GetBoolWithDefault(p.SCtx().GetSessionVars().GetOptimizerFixControlMap(), fixcontrol.Fix56318, true)
```

Comment:
We usually use fix control only in very special cases. For example, this optimization is not easy to use in all scenarios, and other problems may occur after it takes effect.

Your case seems to have limited the scope of the vector function, so fix control is not recommended. If you need a switch to control, it is recommended to use session variable.